pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -1189,6 +1193,124 @@ func (c *Cluster) saveMonConfig() error {
 }
 
 func (c *Cluster) persistExpectedMonDaemons() error {
+	if err := c.persistExpectedMonDaemonsInConfigMap(); err != nil {
+		return errors.Wrap(err, "failed to persist expected mons in ConfigMap")
+	}
+
+	if err := c.persistExpectedMonDaemonsAsEndpointSlice(); err != nil {
+		return errors.Wrap(err, "failed to persist expected mons in EndpointSlice")
+	}
+	return nil
+}
+
+func (c *Cluster) persistExpectedMonDaemonsAsEndpointSlice() error {
+	monitors := c.ClusterInfo.AllMonitors()
+	var monAddressesIPv4, monAddressesIPv6 []string
+
+	if len(monitors) == 0 {
+		logger.Debug("no mon addresses found, skipping endpointslice resource creation")
+		return nil
+	}
+
+	for _, mon := range monitors {
+		ipPort := mon.Endpoint
+
+		var ip string
+		if strings.HasPrefix(ipPort, "[") {
+			ip = ipPort[1:strings.Index(ipPort, "]")]
+		} else {
+			ip = strings.Split(ipPort, ":")[0]
+		}
+
+		parsedIP := net.ParseIP(ip)
+		if parsedIP == nil {
+			logger.Warningf("invalid IP address found: %s", ip)
+			continue
+		}
+
+		if parsedIP.To4() != nil {
+			monAddressesIPv4 = append(monAddressesIPv4, ip)
+		} else {
+			monAddressesIPv6 = append(monAddressesIPv6, ip)
+		}
+	}
+
+	createIPv4 := c.spec.Network.IPFamily == cephv1.IPv4 || c.spec.Network.DualStack
```

Comment:
Why not `!= cephv1.IPv6`, then you don't need the special case below.

```suggestion
	createIPv4 := c.spec.Network.IPFamily != cephv1.IPv6 || c.spec.Network.DualStack
```